File: src/main/java/org/cloudfoundry/credhub/audit/entity/GenerateCredential.java
package org.cloudfoundry.credhub.audit.entity;

import org.cloudfoundry.credhub.audit.OperationDeviceAction;

public class GenerateCredential extends SetCredential {
  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.GENERATE;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/GenerateCredential.kt
package org.cloudfoundry.credhub.audit.entities

import org.cloudfoundry.credhub.audit.OperationDeviceAction

class GenerateCredential : SetCredential() {
    override fun operation(): OperationDeviceAction {
        return OperationDeviceAction.GENERATE
    }
}
